client/src/game/ui/HanabiCard.ts

```diff
@@ -375,15 +388,35 @@ export default class HanabiCard extends Konva.Group implements NodeWithTooltip {
       return UNKNOWN_CARD_RANK;
     }
 
-    // If we have a note identity on the card, show the rank corresponding to the note
-    // (specifically for stack bases in ongoing games; we want notes to have precedence in this case
-    // so that players can make notes in "Throw It in a Hole" variants)
+    // If we have a note on the card and it only provides possibilities of the same rank,
+    // show that rank (specifically for stack bases in ongoing games; we want notes to have
+    // precedence in this case so that players can make notes in "Throw It in a Hole" variants)
+
+    let noteRank = null;
+    if (this.note.possibilities.length !== 0) {
+      const possibleCardsFromNoteAndClues = this.note.possibilities.filter(
+        ([suitIndexA, rankA]) =>
+          this.state.possibleCardsFromClues.findIndex(
+            ([suitIndexB, rankB]) =>
+              suitIndexA === suitIndexB && rankA === rankB,
+          ) !== -1,
+      );
+      const [, candidateRank] = possibleCardsFromNoteAndClues[0];
+      if (
+        possibleCardsFromNoteAndClues.every(
+          ([, rank]) => rank === candidateRank,
```

Comment:
same here

Reply:
unlike the previous case this cannot be replacing with a variable, thus changing this to indexing would remove information from the reader

Reply:
do you insist?

Reply:
> unlike the previous case this cannot be replaced with a variable, thus changing this to indexing would remove information from the reader
then by all means keep it as written = )